org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/DebugProtocol.xtend

```diff
@@ -4173,3 +4236,19 @@ interface InvalidatedAreas {
 	 */
 	public static final String VARIABLES = "variables";
 }
+
+/**
+ * Arguments for 'startDebugging' request.
+ */
+@JsonRpcData
+class StartDebuggingRequestArguments {
+	/**
+	 * Arguments passed to the new debug session. The arguments must only contain
+   * properties understood by the `launch` or `attach` requests of the debug
+   * adapter and they must not contain any client-specific properties (e.g.
+   * `type`) or client-specific features (e.g. substitutable 'variables').
+	 */
+	@NonNull
+	@JsonAdapter(StartDebuggingRequestArgumentsConfigurationTypeAdapter)
+	Either<LaunchRequestArguments, AttachRequestArguments> configuration;
```

Comment:
To make this compatible with `launch` and `attach` perhaps it is best to have this be just a `Map<String, Object>`. Otherwise extenders will all have to provide their own extensions of `StartDebuggingRequestArguments` that understands all the fields.

Note that is change would negate the change needed in the type adapter, but I still would like to see that `request` be marked as a required key in that map, at least through documentation.

Reply:
You're right. I see `launch` and `attach` are not actually using their respective arguments objects. Do you know why that is? Those are the only two requests I see that are like that.

Reply:
Because the argument objects aren't ever useful on their own for `launch` and `attach`, you always need extra fields to make them useful. The spec says "Since launching is debugger/runtime specific, the arguments for this request are not part of this specification." Attach says something similar. I suspect that `restart` and `startDebugging` should say something similar. If you agree I will raise that issue (or add it to https://github.com/microsoft/debug-adapter-protocol/issues/333)

In javascript/typescript it can be typed more easily to the `LaunchRequestArguments` because of the language differences (as I am sure you know). I would be open to any idea that made it better here.